### DISCUSSION TITLE
monitoring dead or stuck connections using Event Exchange plugin

### DISCUSSION THREAD

**Discussion by amichand** [rabbitmq/rabbitmq-server#10326] [Questions]
Hi experts,

I have been using event exchange plugin to detect connections events for managing clients. I noticed that the plugin doesn't generate connection close event if the client was ungracefully terminated or in the case the connection times out due to keepalive miss. Does the broker generate any events for these scenarios that can be consumed via event exchange for monitoring?
Polling for connections list is another way, but I wanted to check if this can be detected via an event.

**Comment by michaelklishin** [comment]
There is a certain reason for that. Due to how the I/O loop is implemented in `rabbit_reader`, emitting events when a connection abruptly terminates would require a pretty drastic (and thus scary/off-putting) amount of refactoring.

One day I'd love us to pull it off but likely not before 3.13 and then 4.0 ship.

**Reply by amichand** [reply]
@michaelklishin thanks for your response.
Whats your recommendation for monitoring dead or timed out connections? 
Any other optimal way other than polling the connection list using rabbitmq management plugin?
